test/identify/index.spec.js

```diff
@@ -430,6 +430,39 @@ describe('Identify', () => {
       await connection.close()
     })
 
+    it('should store remote agent and protocol versions in metadataBook after connecting', async () => {
+      libp2p = new Libp2p({
+        ...baseOptions,
+        peerId
+      })
+
+      await libp2p.start()
+
+      sinon.spy(libp2p.identifyService, 'identify')
+      const peerStoreSpyConsumeRecord = sinon.spy(libp2p.peerStore.addressBook, 'consumePeerRecord')
+      const peerStoreSpyAdd = sinon.spy(libp2p.peerStore.addressBook, 'add')
+
+      const connection = await libp2p.dialer.connectToPeer(remoteAddr)
+      expect(connection).to.exist()
+
+      // Wait for peer store to be updated
+      // Dialer._createDialTarget (add), Identify (consume)
+      await pWaitFor(() => peerStoreSpyConsumeRecord.callCount === 1 && peerStoreSpyAdd.callCount === 1)
+      expect(libp2p.identifyService.identify.callCount).to.equal(1)
+
+      // The connection should have no open streams
+      await pWaitFor(() => connection.streams.length === 0)
+      await connection.close()
+
+      const remotePeer = PeerId.createFromB58String('12D3KooWHFKTMzwerBtsVmtz4ZZEQy2heafxzWw6wNn5PPYkBxJ5')
```

Comment:
Can we do not have the peerId hardcoded?

Reply:
It's taken from https://github.com/libp2p/js-libp2p/blob/master/test/fixtures/browser.js#L6 where it's also hardcoded?

Reply:
I thought I would have been able to get it from [remoteLibp2p](https://github.com/libp2p/js-libp2p/blob/master/test/identify/index.spec.js#L392) but that var does't appear to be set anywhere in the tests?

Reply:
I was meaning, probably more bullet proof to extract the id from the multiaddr

Reply:
Aha, have updated it.